packages/expo-router/src/hooks.ts

```diff
@@ -92,10 +92,27 @@ export function useGlobalSearchParams<
 export function useLocalSearchParams<
   TParams extends SearchParams = SearchParams,
 >(): Partial<TParams> {
-  return (useOptionalLocalRoute()?.params ?? ({} as any)) as Partial<TParams>;
-}
+  const params = React.useContext(NavigationRouteContext)?.params ?? {};
+  return Object.fromEntries(
+    Object.entries(params).map(([key, value]) => {
+      try {
+        if (Array.isArray(value)) {
+          return [
+            key,
+            value.map((v) => {
+              try {
+                return decodeURIComponent(v);
+              } catch {
+                return v;
+              }
+            }),
+          ];
+        }
 
-function useOptionalLocalRoute<T extends RouteProp<ParamListBase>>(): T | undefined {
-  const route = React.useContext(NavigationRouteContext);
-  return route as T | undefined;
+        return [key, decodeURIComponent(value as string)];
+      } catch {
+        return [key, value];
+      }
```

Comment:
The nested try-catch looks unnecessary. Better to write this as:
```ts
if (isArray) {
  try {} catch {}
} else {
  try {} catch {}
}
```
so that unexpected errors don't get caught